File: algorithms/searching/binary-search/python/binary_search.py
def find_item(aList, item):
    start = 0
    end = len(aList) - 1
    while end >= start:
        middle = start + (end - start) // 2
        # if the item at the middle
        if aList[middle] == item:
            return middle
        # if item greater than middle, ignore left half
        if item > aList[middle]:
            start = middle + 1
        else: # if item less than middle, ignore right half
            end = middle - 1

    return -1


myList = [1, 2, 3, 4, 5]
print(find_item(myList, 5))

File: algorithms/searching/binary-search/php/binary-search.php
<?php

function binary_search(array $array, $item){
    $start = 0;
    $end = count($array) - 1;
    while ($end >= $start){
        $middle = $start + ($end - $start) / 2;
        // check if x is present at middle
        if ($array[$middle] == $item){
            return floor($middle);
        }
        // If x greater, ignore left half
        if ($array[$middle] < $item){
            $start = $middle + 1;
        } else { // If x is smaller, ignore right half
            $end = $middle - 1;
        }
    }
    return -1;
}

$myArray = [1, 2, 3, 4, 5, 6];
echo binary_search([1, 2, 3, 4, 5, 6], 4);